### PR TITLE
Add validations and error messages to the activity report

### PR DESCRIPTION
**Pull request by jasalisbury**
**Description of change**
 * The disabled submit buttons are gone
 * If the form is invalid when the submit button is pressed each field that has an error is highlighted
 * The top of the page shows an error that the form is invalid
 * The submitter's draft and needs action review pages show an error if any page of the report is not complete
 * Review/submit pages no longer disable the submit button
 * The review pages show an error if a required field is missing (no approving manager for a submitter or no report status for an approver)
 * Required fields are marked as required
 * Review pages update to use Form Context

**How to test**

1) Pull down, install deps or browse to [sandbox](https://tta-smarthub-sandbox.app.cloud.gov/activity-reports/new/activity-summary)
2) Start a new report. Try to submit the first page. Note the errors.
3) Go to the review page, note each unfinished page is listed as required

**Screenshots**
Screenshot from 2021-02-12 15-19-23
Screenshot from 2021-02-12 15-19-12
Screenshot from 2021-02-12 15-19-07

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/6

**Checklist**

- [x] Meets issue criteria
- [x] Code tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

File: frontend/src/pages/ActivityReport/Pages/topicsResources.js
Comment (dcmcand) [inline frontend/src/pages/ActivityReport/Pages/topicsResources.js:72]: Why did you remove the attachments? According to https://adhoc.slack.com/archives/C016NN6T0Q7/p1611949019010800?thread_ts=1611942489.010400&cid=C016NN6T0Q7 it should stay in.

File: frontend/src/components/Navigator/index.js
Comment (dcmcand) [inline frontend/src/components/Navigator/index.js:163]: It may be out of scope, but I think the submit button should show an alert if the specialist tries to submit a report that has already been submitted and is waiting for approval. Thoughts?

**Comment by kryswisnaskas** [comment]
So far so good. I did run into one error  - when first filling in the dates and then clearing the start date, saw:
image

File: frontend/src/pages/ActivityReport/Pages/topicsResources.js
Comment (jasalisbury) [reply frontend/src/pages/ActivityReport/Pages/topicsResources.js:72]: Oops! I'll add it back in!

File: frontend/src/components/Navigator/index.js
Comment (jasalisbury) [reply frontend/src/components/Navigator/index.js:163]: That'll be taken care of in this [ticket](https://github.com/HHS/Head-Start-TTADP/issues/285)

**Review by dcmcand — APPROVED** [review]
Looks good!

File: frontend/src/pages/ActivityReport/Pages/Review/Submitter/NeedsAction.js
Comment (gopar) [inline frontend/src/pages/ActivityReport/Pages/Review/Submitter/NeedsAction.js:67]: I know we're showing the incomplete pages right above, but would it still make sense to have the button be disabled until the user has completed all the steps?

**Review by gopar — APPROVED** [review]

File: frontend/src/pages/ActivityReport/Pages/Review/Submitter/NeedsAction.js
Comment (jasalisbury) [reply frontend/src/pages/ActivityReport/Pages/Review/Submitter/NeedsAction.js:67]: I'd probably rather it not be disabled so it matches the rest of the form, but I could be convinced otherwise.

Comment (rahearn) [reply frontend/src/pages/ActivityReport/Pages/Review/Submitter/NeedsAction.js:67]: I'm good with this implementation, since the Incomplete pages error info is directly above the button.

**Review by rahearn — CHANGES_REQUESTED** [review]
Really nice work, but I have two changes to request:

1) [I don't think "Target Populations" should be a required field](https://preview.uxpin.com/ad995fd722bd26c0f9a1c8a115f1b41833637308#/pages/134880468/simulate/sitemap)
2) I don't like that the Continue button is enabled but doesn't move the form through the steps if a required field is missing. Currently, if I skip one field and hit continue, it just puts an error message way up at the top of the form off screen.

We should either

* allow continue to navigate through the steps at all times, same as the sidebar (my preference, if you can navigate by the sidebar without completing everything you should be able to use continue as well)
* put the "missing fields" error message directly above the button, so that it is always visible

**Comment by jasalisbury** [comment]
> * allow continue to navigate through the steps at all times, same as the sidebar (my preference, if you can navigate by the sidebar without completing everything you should be able to use continue as well)

If we go that route when do we display validation messages on each page? After they visit the review page, or click the submit button on the review page?

**Comment by jasalisbury** [comment]
> 2\. I don't like that the Continue button is enabled but doesn't move the form through the steps if a required field is missing. Currently, if I skip one field and hit continue, it just puts an error message way up at the top of the form off screen.

Also `react-hook-form` should focus the error when the Continue button is pressed with an invalid form

**Comment by rahearn** [comment]
> Also `react-hook-form` should focus the error when the Continue button is pressed with an invalid form

maybe this isn't working? I hit continue and had no apparent feedback of why I was still looking at the same button.

**Comment by rahearn** [comment]
As discussed here: https://gsa-tts.slack.com/archives/C017RJ4GEET/p1614007348000500 we're going to write a new story to address the validation error on continue and reconciling the differences between navigating with continue buttons vs. the sidenav

**Comment by jasalisbury** [comment]
> As discussed here: https://gsa-tts.slack.com/archives/C017RJ4GEET/p1614007348000500 we're going to write a new story to address the validation error on continue and reconciling the differences between navigating with continue buttons vs. the sidenav

https://github.com/HHS/Head-Start-TTADP/issues/313

**Review by rahearn — APPROVED** [review]
🎉

**Pull request merged**
